### DISCUSSION TITLE
How to pass ECONNREFUSED through the proxy

### DISCUSSION THREAD

**Discussion by martinschayna** [chimurai/http-proxy-middleware#717] [Q&A]
I have a proxy that rewrites requests to couple of servers on the same machine running on different ports. It should be transparent as possible, so when some of target servers aren't running, I want to pass ECONNREFUSED error to the client.

That should be possible inside `onError` handler, default handler looks like that:

```
function onError(err, req, res, target) {
  res.writeHead(504, { 'Content-Type': 'text/plain' });
  res.end('Something went wrong. And we are reporting a custom error message.');
}
```

but obviously, that's not the same, client gives 504 instead of error. What could I put into `onError` handler to simulate original error in client? Thanks.

**Comment by chimurai** [comment]
What you're trying to do, contradicts with the error code `ECONNREFUSED`.

You'll only make debugging more difficult by actually returning a response with the ECONNREFUSED error...

Returning a 502 might be a better practice and log the real ECONNREFUSED for tracing and debugging.

>The HyperText Transfer Protocol (HTTP) 502 Bad Gateway server error response code indicates that the server, while acting as a gateway or proxy, received an invalid response from the upstream server.


https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/502
